core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -324,12 +324,20 @@ trait WhiskMetaApi
             }
         }
 
+        val isRawHTTPEnabled = """(?i)(true)""".r
         extract(_.request.entity.data.length) { length =>
             validateSize(isWhithinRange(length))(transid) {
-                entity(as[Option[JsObject]]) {
-                    body => process(body)
-                } ~ entity(as[FormData]) {
-                    form => process(Some(form.fields.toMap.toJson.asJsObject))
+                optionalHeaderValueByName("x-ow-raw-http") {
+                    case Some(isRawHTTPEnabled(value)) =>
```

Comment:
Is this equivalent to `case Some(headerValue) if headerValue.toBoolean =>`. Using Regex here feels weird.